sklearn/utils/testing.py

```diff
@@ -372,14 +375,52 @@ def __exit__(self, *exc_info):
     assert_greater = _assert_greater
 
 
+def _sparse_dense_allclose(val1, val2, rtol=1e-7, atol=0):
```

Comment:
In reply to @jnothman's comment: this function currently isn't an assert helper, it's just a replacement for `np.allclose` in that it returns True/False. This makes it somewhat not useful for just drop-in simplification of existing tests, I'd prefer an assert helper too. The name of the current function should be `safe_sparse_allclose`, just like `safe_sparse_dot`, we don't need to introduce new naming conventions. The name of the assert helper should probably be `assert_safe_sparse_allclose`, I'm -1 for naming it `assert_allclose` as people who spend more time with sklearn than without it will just start assuming numpy's version is sparse-safe.

I don't think scipy currently has any testing utils. Numpy does, but the sparse matrices are in scipy, so the safe one wouldn't belong in numpy. What might go in scipy is a `scipy.sparse.allclose` function that this could leverage, not a testing assertor, but we can worry about that a bit later, as this one actually should be efficient and not just densifying the input.

The question is: would an `assert_safe_sparse_allclose` actually allow some refactoring of our current tests? @rvraghav93, could you take a look? You can just git-grep for all uses of `assert_allclose` and `assert_array_almost_equal` and see if any are ran on just-densified sparse arrays.